src/main/java/com/nakaligoba/backend/solution/application/SolutionService.java

```diff
@@ -0,0 +1,56 @@
+package com.nakaligoba.backend.solution.application;
+
+import com.nakaligoba.backend.member.domain.Member;
+import com.nakaligoba.backend.member.domain.MemberRepository;
+import com.nakaligoba.backend.problem.domain.Problem;
+import com.nakaligoba.backend.problem.domain.ProblemRepository;
+import com.nakaligoba.backend.programminglanguage.domain.ProgrammingLanguage;
+import com.nakaligoba.backend.programminglanguage.domain.ProgrammingLanguageRepository;
+import com.nakaligoba.backend.solution.controller.dto.SolutionCreateRequest;
+import com.nakaligoba.backend.solution.domain.Solution;
+import com.nakaligoba.backend.solution.domain.SolutionRepository;
+import com.nakaligoba.backend.solutionlanguage.domain.SolutionLanguage;
+import com.nakaligoba.backend.solutionlanguage.domain.SolutionLanguageRepository;
+import com.nakaligoba.backend.utils.JwtUtils;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class SolutionService {
+
+    private final SolutionRepository solutionRepository;
+    private final MemberRepository memberRepository;
+    private final ProblemRepository problemRepository;
+    private final ProgrammingLanguageRepository programmingLanguageRepository;
+    private final SolutionLanguageRepository solutionLanguageRepository;
+
+    public long createSolution(long problemId, SolutionCreateRequest request) {
+        Member member = memberRepository.findByEmail(JwtUtils.getEmailFromSpringSession());
```

Comment:
이렇게 쓰는 것 보다 Controller가 멤버 이메일을 서비스에 전달해주는 것이 좋을 것 같습니다.
전체 애플리케이션에서 봤을 때는 세션에 들어온 멤버가 솔루션을 생성하는 것이지만,
단일 서비스의 메서드 입장에서는 어떤 멤버가 솔루션을 작성하는지 알 수 없습니다.
이는 테스트하기 어려운 메서드가 되구요!
```suggestion
    public long createSolution(String email, long problemId, SolutionCreateRequest request) {
        Member member = memberRepository.findByEmail(email);
```

Reply:
아 맞네요. 단일 서비스 + 테스트 관련해서 말씀하신 부분이 맞네요. 이해했습니다. 감사합니다!